cmd/sources.go

```diff
@@ -0,0 +1,255 @@
+//
+// Copyright (c) 2021 SSH Communications Security Inc.
+//
+// All rights reserved.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/SSHcom/privx-sdk-go/api/rolestore"
+	"github.com/spf13/cobra"
+)
+
+type sourceOptions struct {
+	sourceID string
+}
+
+func init() {
+	rootCmd.AddCommand(sourceListCmd())
+}
+
+//
+//
+func sourceListCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sources",
+		Short: "List and manage user and host directories",
+		Long:  `List and manage user and host directories`,
+		Example: `
+privx-cli sources [access flags]
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return sourceList()
+		},
+	}
+
+	cmd.AddCommand(sourceCreateCmd())
+	cmd.AddCommand(sourceShowCmd())
+	cmd.AddCommand(sourceDeleteCmd())
+	cmd.AddCommand(sourceUpdateCmd())
+	cmd.AddCommand(sourceRefreshCmd())
+
+	return cmd
+}
+
+func sourceList() error {
+	api := rolestore.New(curl())
+
+	sources, err := api.Sources()
+	if err != nil {
+		return err
+	}
+
+	return stdout(sources)
+}
+
+//
+//
+func sourceCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new source",
+		Long:  `Create a new source`,
+		Example: `
+	privx-cli sources create [access flags] JSON-FILE
+		`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return sourceCreate(cmd, args)
+		},
+	}
+
+	return cmd
+}
+
+func sourceCreate(cmd *cobra.Command, args []string) error {
+	var newSource rolestore.Source
+	api := rolestore.New(curl())
+
+	err := decodeJSON(args[0], &newSource)
+	if err != nil {
+		return err
+	}
+
+	id, err := api.CreateSource(newSource)
+	if err != nil {
+		return err
+	}
+
+	return stdout(id)
+}
+
+//
+//
+func sourceShowCmd() *cobra.Command {
+	options := sourceOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Get source by ID",
+		Long:  `Get source by ID. Source ID's are separated by commas when using multiple values, see example`,
+		Example: `
+	privx-cli sources show [access flags] --id <SOURCE-ID>,<SOURCE-ID>
```

Comment:
support only showing a single source (not returning a list). Add a flag to "sources" command to filter by ids as that command returns a list.